### PR TITLE
split the install logic

### PR DESCRIPTION
**Pull request by ZhengzheYang**

### CODE REVIEW

File: asmcli/commands/install.sh [hunk 1]
```diff
@@ -5,8 +5,201 @@ install_subcommand() {
...
+}
+
+start_managed_control_plane() {
+  local PROJECT_ID; PROJECT_ID="$(context_get-option "PROJECT_ID")"
+  local CLUSTER_LOCATION; CLUSTER_LOCATION="$(context_get-option "CLUSTER_LOCATION")"
+  local CLUSTER_NAME; CLUSTER_NAME="$(context_get-option "CLUSTER_NAME")"
+
+  local CR_IMAGE_JSON; CR_IMAGE_JSON="";
+  if [[ -n "${_CI_CLOUDRUN_IMAGE_HUB}" ]]; then
+    CR_IMAGE_JSON=$(cat <<EOF
```
Comment (zerobfd) [inline asmcli/commands/install.sh:32]: Can this just be

```
CR_IMAGE_JSON='{"image": "${_CI_CLOUDRUN_IMAGE_HUB}:${_CI_CLOUDRUN_IMAGE_TAG}"}'
```

**Review by zerobfd — APPROVED** [review]

File: asmcli/commands/install.sh (hunk 1 above)
Comment (ZhengzheYang) [reply asmcli/commands/install.sh:32]: Will the variables be correctly substituted in the single quotes?

Comment (zerobfd) [reply asmcli/commands/install.sh:32]: No of course not. That's silly. This should work though.

```
CR_IMAGE_JSON="{\"image\": \"${_CI_CLOUDRUN_IMAGE_HUB}:${_CI_CLOUDRUN_IMAGE_TAG}\"}"

```

Comment (ZhengzheYang) [reply asmcli/commands/install.sh]: Done!

**Pull request merged**